tools/parmetis_wrapper.py

```diff
@@ -0,0 +1,319 @@
+    num_parts = len(num_nodes_per_chunk[0])
+    ntypes_ntypeid_map, ntypes, ntid_ntype_map = get_node_types(schema_map)
+    type_nid_dict, ntype_gnid_offset = get_idranges(schema_map[constants.STR_NODE_TYPE],
+                                        schema_map[constants.STR_NUM_NODES_PER_CHUNK])
+
+    node_files = []
+    outdir = Path(params.output)
+    os.makedirs(outdir, exist_ok=True)
+    for ntype_id, ntype_name in ntid_ntype_map.items():
+        global_nid_offset = ntype_gnid_offset[ntype_name][0,0]
+        for r in range(num_parts):
+            type_start, type_end = type_nid_dict[ntype_name][r][0], type_nid_dict[ntype_name][r][1]
+            out_file = os.path.join(outdir, 'node_weights_{}_{}.txt'.format(ntype_name, r))
+            node_files.append((out_file, global_nid_offset + type_start, global_nid_offset + type_end))
+
+    nfile = open(os.path.join(params.output, 'parmetis_nfiles.txt'), "w")
+    for f in node_files:
+        #filename global_node_id_start global_node_id_end(exclusive)
+        nfile.write('{} {} {}\n'.format(f[0], f[1], f[2]))
+    nfile.close()
+
+    #Regenerate edge files here...▒
+    edge_data = schema_map[constants.STR_EDGES]
+    edge_files = []
+    for etype_name, etype_info in edge_data.items():
+        edge_info = etype_info[constants.STR_DATA]
+        for r in range(num_parts):
+            out_file = edge_info[r].split("/")[-1]
+            out_file = os.path.join(outdir, 'edges_{}'.format(out_file))
+            edge_files.append(out_file)
+
+    efile = open(os.path.join(params.output, 'parmetis_efiles.txt'), "w")
+    for f in edge_files:
+        efile.write('{}\n'.format(f))
+    efile.close()
+
+
+def run_wrapper(params):
+    print("Starting to generate ParMETIS files...")
+
+    rank, world_size = get_proc_info()
+    schema = read_json(params.schema)
+    num_nodes_per_chunk = schema[constants.STR_NUM_NODES_PER_CHUNK]
+    num_parts = len(num_nodes_per_chunk[0])
+    n = gen_node_weights_files(schema, params.output)
+    print('Done with node weights....')
+
+    e = gen_edge_files(schema, params.output)
+    print('Done with edge weights...')
+
+    if rank == 0:
+        gen_parmetis_input_args(params, schema)
+    print('Done generating files for ParMETIS run ..')
+
+    '''
+    nfiles = []
+    efiles = []
+    for i in range(num_parts):
+        n = gen_node_weights_files(i, schema, params.output)
+        e = gen_edge_files(i, schema, params.output)
+        nfiles.append(n)
+        efiles.append(e)
+
+    nodes = []
+    outer_len = len(nfiles)
+    inner_len = len(nfiles[0])
+    for x in range(inner_len):
+        for y in range(outer_len):
+            nodes.append(nfiles[y][x])
+
+    edges = []
+    outer_len = len(efiles)
+    inner_len = len(efiles[0])
+    for x in range(inner_len):
+        for y in range(outer_len):
+            edges.append(efiles[y][x])
+
+    gen_parmetis_input_args(params, nodes, edges)
+    '''
+
+if __name__ == "__main__":
```

Comment:
pls add tests for this script, just like `//tests/tools/test_dist_part.py`.

Reply:
Added a test as suggested.